src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/DependenciesGraphProvider.cs

```diff
@@ -278,33 +273,26 @@ Graph IGraphProvider.Schema
                     continue;
                 }
 
-                var nodeInfo = inputGraphNode.GetValue<IDependencyNode>(
-                                    DependenciesGraphSchema.DependencyNodeProperty);
+                var nodeInfo = await GetDependencyNodeInfo(graphContext, inputGraphNode, projectPath).ConfigureAwait(false);
                 if (nodeInfo == null)
                 {
                     continue;
                 }
 
-                var subTreeProvider = await GetSubTreeProviderAsync(graphContext, 
-                                                                    inputGraphNode, 
-                                                                    projectPath,
-                                                                    nodeInfo.Id).ConfigureAwait(false);
-                if (subTreeProvider == null)
-                {
-                    continue;
-                }
+                var node = nodeInfo.Item1;
+                var subTreeProvider = nodeInfo.Item2;
 
-                if (!nodeInfo.Flags.Contains(DependencyNode.PreFilledFolderNode))
+                if (!node.Flags.Contains(DependencyNode.PreFilledFolderNode))
```

Comment:
if node has PreFilledFolderNode flag - it is an artificial "folder" item like "framework assemblies", that is having children for sure (otherwise it would not exist), thus we don't need to refresh it and just can set ContainsChildren=true later.